src/services/cache.service.ts

```diff
@@ -30,11 +35,17 @@ export default class CacheService {
     }
   }
 
-  async setSubscription(
-    customerId: string,
-    userType: UserType,
-    subscription: UserSubscription,
-  ): Promise<void> {
+  async getUsedUserPromoCodes(customerId: string): Promise<string[] | null> {
+    const cachedUsedPromoCodesByUser = await this.redis.get(this.buildUsedPromoCodesKey(customerId));
+
+    if (!cachedUsedPromoCodesByUser) {
+      return null;
+    } else {
+      return JSON.parse(cachedUsedPromoCodesByUser) as string[];
```

Comment:
If it is not an object, why `JSON.parse`? And if it is, why `string[]`?

Reply:
Redis only accepts values of type `string`, `Buffer`, or `number` when using the set command (`this.redis.set` in this case).
Therefore, if we want to store an array in Redis, we need to first convert it into a JSON string using `JSON.stringify`,
and then retrieve and parse it using `JSON.parse` to convert it back into an array.

We can safely remove the unnecessary `as string[]` type assertion, but we do need to use `JSON.parse`, otherwise the returned value will remain a plain string.

Reply:
Yeah, I understood that, but the `string[]` type is incorrect then, that is an object, not `string[]`, type it accordingly